server/user/bitrot-detect

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+
+NCPUS=${1}
+if [[ -z "${NCPUS}" ]]; then
+    echo "Missing number of CPUs to use" >&2
+    exit 1
+fi
+
+TARGETDIR=${2}
+if [[ -z "${TARGETDIR}" ]]; then
+    echo "Missing target directory argument" >&2
+    exit 1
+fi
+
+WORKDIR=/var/tmp/bitrot-detect
+mkdir -p ${WORKDIR}
+
+# Clean up previous run.
+rm -f ${WORKDIR}/tbs.md5-*
+
+# We "find" all the controller directories and process them in alphabetical
+# order.
+find ${TARGETDIR}/ -maxdepth 1 -mindepth 1 -type d -printf '%P\n' | sort | while read ctrl; do
+    find ${TARGETDIR}/${ctrl} -type f -name '*.tar.xz' | while read tb; do
```

Comment:
At line 23, you don't need the trailing slash on `${TARGETDIR}/`, and providing it will ensure that `ctrl` ends in a slash, which means that the argument to `find` at line 24` will have a double slash before the final component (and, I believe that that will result in `tb` containing a doubled slash as well...I don't recall).  Not that any of this matters functionally in this case, but there are times when it's annoying, and it's easily avoided.

Reply:
Note that if `${TARGETDIR}/${ctrl}` contains any subdirectories, the `find` at line 24 will descend into them looking for tarballs...is this intentional?  If not, it would be worth adding the `-maxdepth 1` (and presumably `-mindepth 1`, as well) to the invocation.

Reply:
I don't get trailing slashes on subdirs:

```
$ find /srv/pbench/archive/fs-version-001/  -maxdepth 1 -mindepth 1 -type d -printf '%P\n'
a
b
c
```

Reply:
There are only two levels under a controller directory,  and the second levels should not have anything but symlinks, but limiting the `find` is a good thing, particularly after what we saw last month.

Reply:
> I don't get trailing slashes on subdirs

Ah!  That would be the `-printf` -- I'm unaccustomed to that.

> There are only two levels under a controller directory

There are only _supposed_ to be....  I'm a fan of adding resilience and robustness when it doesn't cost too much.  🙂

Reply:
Applied comments.